{{cookiecutter.project_name}}/notebooks/BatchInference.py

```diff
@@ -47,12 +47,26 @@
 assert output_table_name != "", "output_table_name notebook parameter must be specified"
 
 model_name = get_model_name(env)
-model_uri = f"models:/{model_name}/{get_deployed_model_stage_for_env(env)}"
+stage = get_deployed_model_stage_for_env(env)
+model_uri = f"models:/{model_name}/{stage}"
+
+# Get model version from stage
+from mlflow import MlflowClient
+
+model_version_infos = MlflowClient().search_model_versions("name = '%s'" % model_name)
+model_version = next(
```

Comment:
Curious why do we use `next` here?

Reply:
I searched and the only way I found to get the version number of `Production` is to search it. 
The search result is list and I'm using `next` to get the element from it, which is the model version.

Reply:
Ah I see, do we ever expect the size of this list to be > 1 (since multiple versions can be in the same stage), and if so, is it ok to just return the first version?

Reply:
It shouldn't be more than 1 because we are filtering based on the stage. There should be only 1 Production model version and 1 Staging model version.
If this is in the middle of a model deployment and two versions are Production due to race condition, it's fine to use either of them for inference.

Reply:
If alias is in place and there are two `prod` versions, I think the batch should be split into two and each version gets half. It will be a bigger change for inference.

Reply:
Ah well actually we don't disallow two model versions to be in the same stage as of now
<img width="1513" alt="Screen Shot 2023-02-15 at 6 25 32 PM" src="https://user-images.githubusercontent.com/87999496/219252122-cba15dfe-a0d7-4979-834d-acfe1470487c.png">
For example in this case, Version 2 and 3 are both in "Production" but the URI would only return version 3 since it's the latest in the stage.

Reply:
That's surprising. Ok I'll use `max` instead of `next`.
If we are always returning the latest one, we should update to return a random one. Otherwise there's no use of keeping the stage of the old version.